### ISSUE TITLE
thanos-ruler Error no StoreAPIs matched for this query when partial-response is true

### ISSUE THREAD

**Issue by andrejshapal** [thanos-io/thanos#6223]
**Thanos, Prometheus and Golang version used**: go1.19.3 Thanos component version: 0.29.0

**Object Storage Provider**: GCP bucket

**What happened**: When one of our prometheus replicas went down for maintenance, we started to notice absence of any Alertmanager activity. Checking the logs we found:
`	
level=error ts=2023-03-17T12:13:31.023066442Z caller=rule.go:749 component=rules err="read query instant response: expected 2xx response, got 422. Body: {\"status\":\"error\",\"errorType\":\"execution\",\"error\":\"expanding series: proxy Series(): rpc error: code = Aborted desc = No StoreAPIs matched for this query\"}\n" query="(sum by(instance, job, cluster, namespace) (rate(rest_client_requests_total{code=~\"5..\"}[5m])) / sum by(instance, job, cluster, namespace) (rate(rest_client_requests_total[5m]))) > 0.01" `
Basically, thanos-ruler stopped to operate. 
Our settings:
![image](https://user-images.githubusercontent.com/96792836/225911918-fe1d1589-0712-43b3-a0fb-2e5ce96ef85b.png)
![image](https://user-images.githubusercontent.com/96792836/225911986-6f139b5b-6f92-469c-8d3a-0355ffe07405.png)

**What you expected to happen**: Thanos-ruler will give warnings, but keep operating.

**How to reproduce it (as minimally and precisely as possible)**: Set several thanos stores, turn off one and check thanos-ruler logs.

**Environment**:
- OS: k8s
